server/src/main/java/org/opensearch/common/time/JavaDateFormatter.java

```diff
@@ -265,7 +279,7 @@ public TemporalAccessor parse(String input) {
 
         try {
             return doParse(input);
-        } catch (DateTimeParseException e) {
+        } catch (DateTimeException e) {
```

Comment:
Sorry @CaptainDredge , I missed that before - why we are changing exception here? DateTimeFormatter is clearly states that `DateTimeParseException` is raised for parse methods (https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/time/format/DateTimeFormatter.html#parse(java.lang.CharSequence,java.time.temporal.TemporalQuery))

Reply:
@reta we're changing because creation of datetimes through methods like `OffsetDateTime.of` and `ZonedDateTime.atZoneSameInstant` throws `DateTimeException`. `DateTimeParseException` implements `DateTimeException` so all the previous exception cases are as it is covered along with the new ones that are part of this change